diff --git a/drivers/gpu/drm/i915/display/intel_vblank.c b/drivers/gpu/drm/i915/display/intel_vblank.c
index 1b7cfe226ff8..a540615ba431 100644
--- a/drivers/gpu/drm/i915/display/intel_vblank.c
+++ b/drivers/gpu/drm/i915/display/intel_vblank.c
@@ -360,20 +360,16 @@ static bool i915_get_crtc_scanoutpos(struct drm_crtc *_crtc,
 	 * timing critical raw register reads, potentially with
 	 * preemption disabled, so the following code must not block.
 	 */
-	local_irq_save(irqflags);
-	intel_vblank_section_enter(display);
+	preempt_disable();
+	local_irq_save(irqflags);
 
 	/* Get optional system timestamp before query. */
 	if (stime)
 		*stime = ktime_get();
 
 	position = __intel_get_crtc_scanoutpos(crtc, &vpos, &hpos);
-
-	/* Update system timestamp after query. */
-	if (etime)
-		*etime = ktime_get();
 
-	intel_vblank_section_exit(display);
+	local_irq_restore(irqflags);
+	preempt_enable();
 
 	return true;
 }
@@ -764,11 +760,7 @@ int intel_vblank_evade(struct intel_vblank_evade_ctx *evade)
 			break;
 		}
 
-		local_irq_enable();
-
-		timeout = schedule_timeout(timeout);
-
-		local_irq_disable();
+		cpu_relax();
 	}
 
 	finish_wait(wq, &wait);
